.github/workflows/build-docs.yaml

```diff
@@ -1,56 +1,106 @@
+        if: matrix.os == 'ubuntu-latest'
         run: |
           echo `env`
-          git init .
-          git remote add origin https://github.com/mozilla/application-services.git
-          git pull origin main
-          git submodule init
-          git submodule update --recursive
           echo 'rust.targets=linux-x86-64' > local.properties
           curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
           source $HOME/.cargo/env
- 
-      - name: Build Dependencies
-        env: 
+
+      - name: Set up Rust (macOS only)
+        if: matrix.os == 'macos-latest'
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Set up Ruby (macOS only)
+        if: matrix.os == 'macos-latest'
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.3'
+
+      - name: Install Jazzy (macOS only)
+        if: matrix.os == 'macos-latest'
+        run: gem install jazzy
+
+      - name: Build Dependencies (Ubuntu only)
+        if: matrix.os == 'ubuntu-latest'
+        env:
           NSS_DIR: ${{ github.workspace }}/libs/desktop/linux-x86-64/nss
-          NSS_STATIC: 1 
+          NSS_STATIC: 1
         run: |
-
           git clone https://chromium.googlesource.com/external/gyp.git tools/gyp
           cd tools/gyp
           sudo python setup.py install
           cd ../..
           sudo apt-get install ninja-build zlib1g-dev tclsh python3
           ./libs/verify-desktop-environment.sh
-      
-      - name: Build mdbook
-        env: 
+
+      - name: Build mdbook (Ubuntu only)
+        if: matrix.os == 'ubuntu-latest'
+        env:
           NSS_DIR: ${{ github.workspace }}/libs/desktop/linux-x86-64/nss
-          NSS_STATIC: 1 
+          NSS_STATIC: 1
         run: |
           cargo install mdbook mdbook-mermaid mdbook-open-on-gh
           ./tools/build-book.sh
-          
-      - name: Deploy
+
+      - name: Build Swift docs (macOS only)
+        if: matrix.os == 'macos-latest'
+        run: |
+          cd ./automation/swift-components-docs
+          ./build.sh
+
+      - name: Upload docs artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: docs-${{ matrix.os }}
+          path: |
+            ${{ matrix.os == 'ubuntu-latest' && 'build/docs' || 'automation/swift-components-docs/docs' }}
+
+  deploy-docs:
+    needs: build-docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Download all artifacts
+        uses: actions/download-artifact@v3
+
+      - name: Prepare documentation
+        run: |
+          mkdir -p gh-pages
+          mv docs-ubuntu-latest/* gh-pages/
+          mkdir -p gh-pages/swift
+          mv docs-macos-latest/* gh-pages/swift/
+
+      - name: Deploy to GitHub Pages
         run: |
-          git config --local user.email "${{ github.actor }}@users.noreply.github.com"
```

Comment:
Not sure if we want to leave it as a local config, or set it to a github actions bot.

Reply:
I don't know GitHub actions well enough to have an opinion.   I'd say go with whatever you think is the best practice.